splunk-otel-android/src/test/java/com/splunk/rum/volley/TracingHurlStackTest.java

```diff
@@ -0,0 +1,274 @@
+        shadowOf(getMainLooper()).idle();
+
+        String result = response.get(10, TimeUnit.SECONDS);
+
+        assertThat(server.takeRequest().getPath()).isEqualTo("/success"); //server received request
+        assertThat(result).isEqualTo("success");
+
+        List<SpanData> spans = exporter.getFinishedSpanItems();
+        assertThat(spans).hasSize(1);
+
+        SpanData span = spans.get(0);
+
+        verifyAttributes(span, url, 200);
+
+    }
+
+    @Test
+    public void serverError() throws IOException, InterruptedException {
+
+        server.enqueue(new MockResponse().setResponseCode(500));
+        server.play();
+
+        String url = server.getUrl("/error").toString();
+
+        RequestFuture<String> response = RequestFuture.newFuture();
+        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
+                response, response);
+
+        testQueue.addToQueue(stringRequest);
+
+        Thread.sleep(100); // Volley works asynchronously, we need to wait for Looper to receive the task
+        shadowOf(getMainLooper()).idle();
+
+        assertThatThrownBy(() -> response.get(10, TimeUnit.SECONDS)).hasCauseInstanceOf(VolleyError.class);
+
+        assertThat(server.takeRequest().getPath()).isEqualTo("/error"); //server received request
+
+        List<SpanData> spans = exporter.getFinishedSpanItems();
+        assertThat(spans).hasSize(1);
+
+        SpanData span = spans.get(0);
+
+        verifyAttributes(span, url, 500);
+    }
+
+    @Test
+    public void connectionError() throws IOException, InterruptedException {
+
+        server.enqueue(new MockResponse().setBody("should not be received"));
+        server.play();
+
+        String url = "http://" + server.getHostName() + ":" + findUnusedPort() + "/none";
+
+        RequestFuture<String> response = RequestFuture.newFuture();
+
+        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
+                response, response);
+        stringRequest.setRetryPolicy(new DefaultRetryPolicy(100, 0, 1f));
+
+        testQueue.addToQueue(stringRequest);
+
+        Thread.sleep(1000); // Volley works asynchronously, we need to wait for Looper to receive the task
+        shadowOf(getMainLooper()).idle();
+
+        assertThatThrownBy(() -> response.get(3, TimeUnit.SECONDS)).hasCauseInstanceOf(TimeoutError.class);
+
+        assertThat(server.getRequestCount()).isEqualTo(0); //server received no requests
+
+        List<SpanData> spans = exporter.getFinishedSpanItems();
+        assertThat(spans).hasSize(1);
+
+        SpanData span = spans.get(0);
+        assertThat(span.getName()).isEqualTo("HTTP GET");
+
+        assertThat(span.getStatus()).isEqualTo(StatusData.error());
+
+        EventData event = span.getEvents().stream().filter(e -> e.getName().equals(SemanticAttributes.EXCEPTION_EVENT_NAME)).findFirst().orElse(null);
+
+        assertThat(event).isNotNull();
```

Comment:
You could probably use some AssertJ idiom to assert on that:
```suggestion
        assertThat(span.getEvents())
            .hasSize(1)
            .allSatisfy(e -> e.getName().equals(SemanticAttributes.EXCEPTION_EVENT_NAME))
```